### PR TITLE
Prevent investigate request timing out

### PR DESCRIPTION
**Pull request by okurz**
Add timeout for "git diff" call in investigate to prevent the complete
request to timeout. Calculating "git diff" can sometimes take very long
compared to the other parts of the investigation process.

### CODE REVIEW

**Comment by okurz** [comment]
Need to test this more. An initial manual test yields now always "an internal error occured".

**Comment by Martchus** [comment]
As you mention that there are problems it comes to my mind that we usually change `OpenQA::Setup::read_config` for such changes. That's also the global place where the default value should go to.

**Comment by okurz** [comment]
Well, the problem was something different. apparmor :)

Fixed that. I wanted to discuss the config handling in another scope anyway if you don't mind. I guess it's easy for us to grep for readout from the config files from our whole code later.

**Comment by okurz** [comment]
Fine. So I updated `OpenQA::Setup::read_config` and the template config file accordingly.

File: lib/OpenQA/Schema/Result/Jobs.pm [hunk 1]
```diff
@@ -1837,7 +1837,8 @@ sub git_log_diff {
 
 sub git_diff {
     my ($self, $dir, $refspec_range) = @_;
-    my $res = run_cmd_with_log_return_error(['git', '-C', $dir, 'diff', '--stat', $refspec_range]);
+    my $timeout = OpenQA::App->singleton->config->{global}->{job_investigate_git_timeout} // 20;
```
Comment (Martchus) [inline lib/OpenQA/Schema/Result/Jobs.pm:1840]: You don't need the 20 here. But it is already at 3 other places so I don't care.

**Pull request merged**